### PR TITLE
xarray_utils: calculate global mean

### PR DESCRIPTION
**Pull request by mathause**
- [ ] Closes #xxx
 - [x] Tests added
 - [x] Passes `isort . && black . && flake8`
 - [x] Fully documented, including `CHANGELOG.rst`

Next data handling PR: calculating weighted global means.

### CODE REVIEW

**Comment by mathause** [comment]
@znicholls: as always a quick look and your opinion on naming is appreciated.

File: docs/source/api.rst [hunk 1]
```diff
@@ -36,6 +36,15 @@ Computation
    ~core.computation.calc_geodist_exact
    ~core.computation.gaspari_cohn
 
+Data manipulation
+-----------------
+
+.. autosummary::
+   :toctree: generated/
+
+   ~xarray_utils.globmean.lat_weights
```
Comment (znicholls) [inline docs/source/api.rst:45]:
```suggestion
   ~xarray_utils.global_mean.lat_weights
```

I'd be tempted to just write it out, but doesn't really matter

File: tests/unit/test_globmean.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+def test_lat_weights_2D_warn_2D():
+
+    lat = np.arange(10).reshape(2, 5)
+
+    with pytest.warns(UserWarning, match="non-regular grids"):
+        mxu.globmean.lat_weights(lat)
+
+
+@pytest.mark.parametrize("lat", [-91, 90.1])
+def test_lat_weights_2D_warn_90(lat):
```
Comment (znicholls) [inline tests/unit/test_globmean.py:61]:
```suggestion
def test_lat_weights_2D_error_90(lat):
```

File: tests/unit/test_globmean.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+@pytest.mark.parametrize("as_dataset", [True, False])
+def test_calc_globmean_default(as_dataset):
+
+    _test_calc_globmean(as_dataset)
+
+
+@pytest.mark.parametrize("as_dataset", (True, False))
+@pytest.mark.parametrize("x_dim", ("x", "lon"))
+@pytest.mark.parametrize("y_dim", ("y", "lat"))
+def test_ocean_land_fraction(as_dataset, x_dim, y_dim):
```
Comment (znicholls) [inline tests/unit/test_globmean.py:101]: How does this check the fraction?

**Review by znicholls — APPROVED** [review]
lgtm, minor comments.

Should be easy enough to extend to using model reported grid cell areas, should we ever go down that path...

File: tests/unit/test_globmean.py (hunk 3 above)
Comment (mathause) [reply tests/unit/test_globmean.py:101]: Good catch - thanks!

**Pull request merged**